spec/models/instrument_price_policy_spec.rb

```diff
@@ -175,4 +175,80 @@
     end
   end
 
+  describe "daily rates" do
+    let(:instrument) do
+      build :instrument, pricing_mode: Instrument::Pricing::SCHEDULE_DAILY
+    end
+
+    let(:policy) { build :instrument_price_policy, product: instrument }
+
+    shared_examples "validates policy field" do |field_name|
+      before { policy.valid? }
+
+      it { expect(policy).to_not be_valid }
+      it { expect(policy.errors).to include(field_name) }
+    end
+
+    describe "with negative usage_rate_daily value" do
+      before { policy.usage_rate_daily = -1 }
+
+      it_behaves_like "validates policy field", :usage_rate_daily
+    end
+
+    describe "with negative usage_subsidy_daily value" do
+      before { policy.usage_subsidy_daily = -1 }
+
+      it_behaves_like "validates policy field", :usage_subsidy_daily
+    end
+
+    describe "with negative usage_rate_daily value" do
+      before { policy.usage_rate_daily = -1 }
+
+      it_behaves_like "validates policy field", :usage_rate_daily
+    end
+
+    describe "with missing usage_rate_daily value" do
+      before { policy.usage_rate_daily = nil }
+
+      it_behaves_like "validates policy field", :usage_rate_daily
+    end
+
+    describe "validates subsidy equals to rate" do
```

Comment:
I think we're testing this case

```suggestion
    describe "validates subsidy greater than rate" do
```